vimms/Controller/misc.py

```diff
@@ -11,140 +8,27 @@ class FixedScansController(Controller):
     A controller which takes a schedule of scans, converts them into tasks in queue
     """
 
-    def __init__(self, ionisation_mode, schedule,
-                 isolation_width=DEFAULT_ISOLATION_WIDTH,
-                 params=None):
+    def __init__(self, schedule, params=None):
         super().__init__(params=params)
-
-        self.ionisation_mode = ionisation_mode
-        self.isolation_width = isolation_width
-        self.schedule = schedule
+        if len(schedule) > 0:
+            self.set_tasks(schedule)
 
     def get_initial_tasks(self):
-        if isinstance(self.schedule, list):  # allows sending list of tasks
-            tasks = self.schedule
-        else:
-            tasks = self._load_tasks(self.schedule)
-        return tasks
+        return self.tasks
```

Comment:
Suggest an assertion in here -- so that it squawks nicely if we get to this point and no tasks have actually been set. I.e. user doesn't set them in the constructor and then sets then with `set_tasks` later

Reply:
Added in commit https://github.com/sdrogers/vimms/commit/764e9ce741f22dc2b00f864a49d077e9c7151e82